### DISCUSSION TITLE
Little functionality from the user-side

### DISCUSSION THREAD

**Discussion by peekxc** [marp-team/marp#210] [Q&A] [answered]
On one hand, Marpit seems to offer a clean, minimal, and extensible solution to quickly building markdown-based slide-decks. 

This aspect of the Marp ecosystem seem very well developed, and the documentation is also very good. Well done!

---

However, from a user standpoint (my own), the functionality Marp core provides seems a bit limited without resulting to custom CSS and Node-based extensions, etc. For example, I really like the idea of just learning a basic set of [directives](https://marpit.marp.app/directives) which can be used to build 95% of the things one would want out of a nice slide deck... but the only directives so far in marp core are: 

**global:** theme, style, headingDivider
**local:** paginate, header, footer, class, background*, color

LaTeX is handled via its respective engine (KaTeX, MathJAX), Markdown via its respective engine (Commonmark, etc), and emoji via whatever its rendering engine is... and that's it! This seems quite minimal!

For example, some would consider [Deckset](https://docs.decksetapp.com/English.lproj/getting-started.html) quite "minimal" in its feature set, but it has a number of utilities supported out of the box which are great! Step-by-step code highlighting, footnotes, multi-column support, non-pure-CSS-driven image manipulation commands, etc. 

So my question is whether Marp is intended to be as minimalistic as it is with the directives _by design_ or whether its waiting for the developer community to make enough CSS classes for it to be more feature-rich for users who don't want to code everything from scratch or....? Many of the things I mentioned it seems to me are directives that could easily be added to Marp core (indeed, thanks to Marps great extensibility!)... so why aren't they there?

I understand that maybe the goal is to make an unopinionated framework, but even unopinionated libraries like [Tailwind](https://tailwindcss.com/) have a bunch of utility classes to bootstrap a custom layout. 

Am I missing something?

**Comment by yhatt** [comment, marked as answer]
> whether Marp is intended to be as minimalistic as it is with the directives _by design_

Strongly YES. Marp ecosystem _must be minimal toolset._

> whether its waiting for the developer community to make enough CSS classes for it to be more feature-rich for users who don't want to code everything from scratch

For long-live maintainability (and keeping compatibillity with standard), we are not active for adding more rich features. It has already mentioned in [our contributing guideline](https://github.com/marp-team/.github/blob/master/CONTRIBUTING.md#before-adding-new-feature). Even though adding directives and CSS layouts is easy, keeping maintenance ever is very hard work for us.
